internal/provider/resource_auth_method_oidc.go

```diff
@@ -0,0 +1,515 @@
+	}
+	if amcr == nil {
+		return diag.Errorf("nil auth method after create")
+	}
+	// need to update state is create was successful to avoid loosing state if the change state or update
+	// scope step that follows errors out
+	if err := setFromOidcAuthMethodResponseMap(d, amcr.GetResponse().Map); err != nil {
+		return diag.Errorf("%v", err)
+	}
+
+	amid := amcr.GetResponse().Map["id"].(string)
+
+	// auto set to active-public state
+	_, err = amClient.ChangeState(ctx, amid, 0, "active-public", authmethods.WithAutomaticVersioning(true))
+	if err != nil {
+		return diag.Errorf("%v", err)
+	}
+
+	// update scope when set to primary
+	if p, ok := d.GetOk(authmethodOidcIsPrimaryAuthMethodForScope); ok {
+		if p.(bool) {
+			if err := updateScopeWithPrimaryAuthMethodId(ctx, scopeId, amid, meta); err != nil {
+				return diag.Errorf("%v", err)
+			}
+
+			amcr.GetResponse().Map[authmethodOidcIsPrimaryAuthMethodForScope] = true
+		}
+	}
+
+	return setFromOidcAuthMethodResponseMap(d, amcr.GetResponse().Map)
+}
+
+func updateScopeWithPrimaryAuthMethodId(ctx context.Context, scopeId, authmethodId string, meta interface{}) diag.Diagnostics {
+	md := meta.(*metaData)
+	scp := scopes.NewClient(md.client)
+
+	opts := []scopes.Option{}
+	opts = append(opts, scopes.WithAutomaticVersioning(true))
+	opts = append(opts, scopes.WithPrimaryAuthMethodId(authmethodId))
+
+	_, err := scp.Update(ctx, scopeId, 0, opts...)
+	if err != nil {
+		return diag.Errorf("error updating scope: %v", err)
+	}
+
+	return nil
+}
+
+func readScopeIsPrimaryAuthMethodId(ctx context.Context, scopeId, authmethodId string, meta interface{}) (diag.Diagnostics, bool) {
+	md := meta.(*metaData)
+	scp := scopes.NewClient(md.client)
+
+	srr, err := scp.Read(ctx, scopeId)
+	if err != nil {
+		return diag.Errorf("%s", err), false
+	}
+
+	if p, ok := srr.GetResponse().Map["primary_auth_method_id"]; ok {
+		if p == authmethodId {
+			return nil, true
+		}
+	}
+
+	return nil, false
+}
+
+func resourceAuthMethodOidcRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	md := meta.(*metaData)
+	amClient := authmethods.NewClient(md.client)
+
+	amrr, err := amClient.Read(ctx, d.Id())
+	if err != nil {
+		if apiErr := api.AsServerError(err); apiErr.Response().StatusCode() == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("error reading auth method: %v", err)
+	}
+	if amrr == nil {
+		return diag.Errorf("auth method nil after read")
+	}
+
+	serr, isPrimary := readScopeIsPrimaryAuthMethodId(ctx, amrr.GetResponse().Map["scope_id"].(string), amrr.GetResponse().Map["id"].(string), meta)
+	if err != nil {
+		return diag.Errorf("%v", serr)
+	}
+
+	if isPrimary {
+		amrr.GetResponse().Map[authmethodOidcIsPrimaryAuthMethodForScope] = true
+	}
+
+	return setFromOidcAuthMethodResponseMap(d, amrr.GetResponse().Map)
+}
+
+func resourceAuthMethodOidcUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
future PR idea for both Create/Update: use a common func to build the opts.  Not complete but something like:
```
func buildOpts(d *schema.ResourceData, fields []string) []authmethods.Option {
	opts := []authmethods.Option{}

	m := map[string]func() authmethods.Option{
		NameKey: func() authmethods.Option {
			val, ok := d.GetOk(NameKey)
			if ok {
				return authmethods.WithName(val.(string))
			}
			return authmethods.DefaultName()
		},
		DescriptionKey: func() authmethods.Option {
			val, ok := d.GetOk(DescriptionKey)
			if ok {
				return authmethods.WithName(val.(string))
			}
			return nil
		},
	}
	for _, fieldName := range fields {
		if o := m[fieldName](); o != nil {
			opts = append(opts, o)
		}
	}
	return opts
}
```